bin/peggy.mjs

```diff
@@ -0,0 +1,331 @@
+  }
+
+  const deps = {};
+  for (const dep of options.dependency) {
+    const [name, val] = dep.split(":");
+    deps[name] = val ? val : name;
+  }
+  options.dependencies = deps;
+  delete options.dependency;
+}
+
+if (options.exportVar !== undefined) {
+  if ((options.format !== "globals") && (options.format !== "umd")) {
+    abort(`Can't use the -e/--export-var option with the "${options.format}" module format.`);
+  }
+}
+
+let inputFile = options.input;
+switch (program.args.length) {
+  case 0:
+    inputFile = inputFile ? inputFile : "-";
+    break;
+
+  case 1:
+    if (inputFile) {
+      abort("Do not specify input both on command line and in config file.");
+    }
+    inputFile = program.args[0];
+    break;
+
+  default:
+    abort("Too many arguments.");
+}
+
+let outputFile = options.output;
+if (!outputFile) {
+  outputFile = ((inputFile === "-") || options.test || options.testFile)
+    ? outputFile = "-"
+    : inputFile.substr(0, inputFile.length - path.extname(inputFile).length) + ".js";
+}
+
+if (options.test && options.testFile) {
+  abort("The -t/--test and -T/--test-file options are mutually exclusive.");
+}
+
+options.output = (options.test || options.testFile) ? "parser" : "source";
+let testText = null;
+let testGrammarSource = null;
+if (options.test) {
+  testText = options.test;
+  testGrammarSource = "command line";
+  delete options.test;
+}
+if (options.testFile) {
+  testText = readFile(options.testFile);
+  testGrammarSource = options.testFile;
+  delete options.testFile;
+}
+
+if (verbose) {
+  console.error("OPTIONS:", util.inspect(options, {
+    depth: Infinity,
+    colors: process.stdout.isTTY,
+  }));
+  console.error("INPUT:", inputFile);
+  console.error("OUTPUT:", outputFile);
+  console.error("TEST TEXT:", testText);
+}
+
+// Main
+
+let inputStream, outputStream;
+
+if (inputFile === "-") {
+  process.stdin.resume();
+  inputStream = process.stdin;
+  inputStream.on("error", () => {
+    abort("Can't read from stdin.");
+  });
+  options.grammarSource = "stdin";
+} else {
+  options.grammarSource = inputFile;
+  inputStream = fs.createReadStream(inputFile);
+}
+
+readStream(inputStream, input => {
+  let source;
+
+  try {
+    source = peg.generate(input, options);
+  } catch (e) {
+    abortError("Error parsing grammar:", e);
+  }
+
+  if (testText) {
+    try {
+      source = source.parse(testText, {
+        grammarSource: testGrammarSource,
+      });
+    } catch (e) {
+      abortError("Error parsing test:", e);
+    }
+    source = util.inspect(source, {
+      depth: Infinity,
+      colors: (outputFile === "-") && process.stdout.isTTY,
+    }) + "\n";
+  }
+
+  // Don't create output until processing succeeds
+  if (outputFile === "-") {
+    outputStream = process.stdout;
+  } else {
+    outputStream = fs.createWriteStream(outputFile);
+    outputStream.on("error", () => {
+      abort("Can't write to file \"" + outputFile + "\".");
```

Comment:
```suggestion
      abort(`Can't write to file "${outputFile}".`);
```

Reply:
Willfix, along with other comments I've marked as "resolved" without commenting on them.  I wonder if we should add `prefer-template`to the lint file.